src/main/java/digital/slovensko/autogram/core/Autogram.java

```diff
@@ -60,27 +67,83 @@ public void sign(SigningJob job, SigningKey signingKey) {
                 job.signWithKeyAndRespond(signingKey);
                 ui.onUIThreadDo(() -> ui.onSigningSuccess(job));
             } catch (DSSException e) {
-                ui.onUIThreadDo(() -> ui.onSigningFailed(AutogramException.createFromDSSException(e)));
+                onSigningFailed(AutogramException.createFromDSSException(e));
             } catch (IllegalArgumentException e) {
-                ui.onUIThreadDo(() -> ui.onSigningFailed(AutogramException.createFromIllegalArgumentException(e)));
+                onSigningFailed(AutogramException.createFromIllegalArgumentException(e));
             } catch (Exception e) {
-                ui.onUIThreadDo(() -> ui.onSigningFailed(new UnrecognizedException(e)));
+                onSigningFailed(new UnrecognizedException(e));
             }
         });
     }
 
+    /**
+     * Starts a batch - ask user - get signing key - start batch - return batch ID
+     * 
+     * @param totalNumberOfDocuments - expected number of documents to be signed
+     * @param responder              - callback for http response
+     */
+    public void batchStart(int totalNumberOfDocuments, BatchResponder responder) {
+        if (batch != null && !batch.isEnded())
+            throw new BatchConflictException("Iné hromadné podpisovanie už prebieha");
+        batch = new Batch(totalNumberOfDocuments);
+
+        var startBatchTask = new AutogramBatchStartCallback(batch, responder);
+
+        ui.onUIThreadDo(() -> {
+            ui.startBatch(batch, this, startBatchTask);
+        });
+    }
+
+    /**
+     * Sign a single document
+     * 
+     * @param job
+     * @param batchId - current batch ID, used to authenticate the request
+     */
+    public void batchSign(SigningJob job, String batchId) {
+        if (batch == null)
+            throw new BatchNotStartedException();
```

Comment:
Dajme tu checked exception ak sa da, kedze je to best practice podla @sabomichal

Reply:
ak tomu dobre rozumiem, tak by bolo treba prerobit `AutogramException` na interface, a urobit `AutogramRuntimeException extends RuntimeException` (unchecked)  a `AutogramCheckedException extends Exception` (checked)

mozem to urobit, ale je to dost velka zmena (skoro vsetky classy), tak neviem ci to patri do tohoto PR

Reply:
Ok, nechame na neskor.